pull.py

```diff
@@ -103,8 +170,83 @@
+    'GPLv2': {'spdx': 'GPL-2.0'},
+    'HPND': {'spdx': 'HPND'},
+    'IBMPL': {'spdx': 'IPL-1.0'},
+    'iMatix': {'spdx': 'iMatix'},
+    'imlib': {'spdx': 'Imlib2'},
+    'ijg': {'spdx': 'IJG'},
+    'intel': {'spdx': 'Intel'},
+    'IPAFONT': {'spdx': 'IPA'},
+    'ISC': {'spdx': 'ISC'},
+    'JSON': {'spdx': 'JSON'},
+    'LAL-1.2': {'spdx':'LAL-1.2'},
+    'LAL-1.3': {'spdx':'LAL-1.3'},
+    'LGPLv3': {'spdx': 'LGPL-3.0'},
+    'LGPLv2.1': {'spdx': 'LGPL-2.1'},
+    'LPPL-1.2': {'spdx': 'LPPL-1.2'},
+    'LPPL-1.3a': {'spdx': 'LPPL-1.3a'},
+    'lucent102': {'spdx': 'LPL-1.02'},
+    'ModifiedBSD': {'spdx': 'BSD-3-Clause'},
+    'MPL': {'spdx': 'MPL-1.1'},
+    'MPL-2.0': {'spdx':'MPL-2.0'},
+    'ms-pl': {'spdx': 'MS-PL'},
+    'ms-rl': {'spdx': 'MS-RL'},
+    'NASA': {'spdx': 'NASA-1.3'},
+    'NCSA': {'spdx':'NCSA'},
+    'newOpenLDAP': {'spdx': 'OLDAP-2.7'},
+    'Nokia': {'spdx': 'Nokia'},
+    'NoLicense': {'spdx': 'NONE'},
+    'NOSL': {'spdx': 'NOSL'},
+    'NPL-1.0': {'spdx': 'NPL-1.0'},
+    'NPL-1.1': {'spdx': 'NPL-1.1'},
+    'ODbl': {'spdx': 'ODbL-1.0'},
+    'oldOpenLDAP': {'spdx': 'OLDAP-2.7'},
+    'OpenPublicL': {'spdx': 'OPL-1.0'},
+    'OpenSSL': {'spdx': 'OpenSSL'},
+    'OriginalBSD': {'spdx': 'BSD-4-Clause'},
+    'OSL-1.0': {'spdx': 'OSL-1.0'},
+    'OSL-1.1': {'spdx': 'OSL-1.1'},
+    'OSL-2.0': {'spdx': 'OSL-2.0'},
+    'OSL-2.1': {'spdx': 'OSL-2.1'},
+    'OSL-3.0': {'spdx': 'OSL-3.0'},
+    'PHP-3.01': {'spdx': 'PHP-3.01'},
+    'Python': {'spdx': 'Python-2.0'}, # Note: references 'later versions which are not in the SPDX license list
+    'QPL': {'spdx': 'QPL-1.0'},
+    'RPL-1.1': {'spdx': 'RPL-1.1'},
+    'RPL-1.5': {'spdx': 'RPL-1.5'},
+    'RPSL': {'spdx': 'RPSL-1.0'},
+    'Ruby': {'spdx': 'Ruby'}, # Note that the text linked is 'not an exact match to the SPDX license list
+    'SGIFreeB': {'spdx': 'SGI-B-2.0'},
+    'SILOFL': {'spdx': 'OFL-1.1'},
+    'SISSL': {'spdx': 'SISSL'}, # Note that the header on the 'FSF website states version 1.0, but the link points to 'version 1.1.  The SPDX license is version 1.1
+    'SPL': {'spdx': 'SPL-1.0'},
+    'StandardMLofNJ': {'spdx': 'SMLNJ'},
+    'Unicode-DFS-2015': {'spdx': 'Unicode-DFS-2015'},
+    'Unicode-DFS-2016': {'spdx': 'Unicode-DFS-2016'},
+    'Unlicense': {'spdx': 'Unlicense'},
+    'UPL': {'spdx': 'UPL-1.0'},
+    'Vim': {'spdx': 'Vim'},
+    'W3C': {'spdx': 'W3C'},
+    'W3C-20150513': {'spdx': 'W3C-20150513'},
+    'W3C-19980720': {'spdx': 'W3C-19980720'},
+    'Watcom': {'spdx': 'Watcom-1.0'},
+    'WTFPL': {'spdx': 'WTFPL'},
+    'X11License': {'spdx': 'X11'},
+    'XFree861.1License': {'spdx': 'XFree86-1.1'},
+    'xinetd': {'spdx': 'xinetd'},
+    'Yahoo': {'spdx': 'YPL-1.1'},
+    'Zend': {'spdx': 'Zend-2.0'},
+    'Zimbra': {'spdx': 'Zimbra-1.3'},
+    'ZLib': {'spdx': 'Zlib'},
+    'Zope': {'spdx': 'ZPL-1.1'}, # Note the FSF refers to version 1.0 and SPDX uses version 1.1 - it should be verified that 1.1 should be included
```

Comment:
I'd rather leave these inexact matches off.  If someone is looking up GPL-compat in this API, for example, false negatives are much less problematic than false positives.

Reply:
If we want to take a more conservative approach, I would remove the SPDX ID's for Zope and Ruby.  I consider SISSL to be an internal inconsistency on the FSF website and I believe they intend it to match the referenced SPDX license.

In your review comment, you included Zlib, Zimbra and Zend.  I'm assuming your comment only referred to Zope - let me know that is incorrect or if there are other license matches that concern you.

Reply:
> In your review comment, you included Zlib, Zimbra and Zend. I'm assuming your comment only referred to Zope - let me know that is incorrect or if there are other license matches that concern you.

I'd rather have this bulk PR only cover exact matches.  Can you drop anything you consider questionable and file those in per-license(-group) PRs?

Reply:
Done - I already removed the matches I consider questionable and added them to the list of FSF license ids without an associated SPDX license ID.